db/attachment_compaction.go

```diff
@@ -29,102 +29,115 @@ const (
 	CleanupPhase    = "cleanup"
 )
 
-func attachmentCompactMarkPhase(ctx context.Context, dataStore base.DataStore, collectionID uint32, db *Database, compactionID string, terminator *base.SafeTerminator, markedAttachmentCount *base.AtomicInt) (count int64, vbUUIDs []uint64, checkpointPrefix string, err error) {
-	base.InfofCtx(ctx, base.KeyAll, "Starting first phase of attachment compaction (mark phase) with compactionID: %q", compactionID)
-	compactionLoggingID := "Compaction Mark: " + compactionID
-
-	var markProcessFailureErr error
-
-	// failProcess used when a failure is deemed as 'un-recoverable' and we need to abort the compaction process.
-	failProcess := func(err error, format string, args ...interface{}) bool {
-		markProcessFailureErr = err
-		terminator.Close()
-		base.WarnfCtx(ctx, format, args...)
-		return false
+// processAttachmentCompactSweepCallback(ctx context.Context, dataStore base.DataStore, db *Database, compactionID, compactionLoggingID string, dryRun bool, purgedAttachmentCount *base.AtomicInt, event sgbucket.FeedEvent) {
+func processAttachmentCompactMarkCallback(ctx context.Context, dataStore base.DataStore, compactionID, compactionLoggingID string, markedAttachmentCount *base.AtomicInt, event sgbucket.FeedEvent) error {
```

Comment:
As discussed offline, I removed these changes because they were bigger than was appropriate. I left a single change to treat a document with an invalid xattr as a document with xattrs.